crates/router/src/connector/square/transformers.rs

```diff
@@ -181,7 +178,7 @@ impl TryFrom<&types::TokenizationRouterData> for SquareTokenRequest {
                 Self::try_from((item, pay_later_data))
             }
             api::PaymentMethodData::GiftCard(_) => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
+                utils::get_unimplemented_payment_method_error_message("Square"),
```

Comment:
We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice